spark/src/main/kotlin/com/adevinta/spark/res/AnnotatedStringResource.kt

```diff
@@ -0,0 +1,238 @@
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.LocalDensity
+import androidx.compose.ui.text.AnnotatedString
+import androidx.compose.ui.text.SpanStyle
+import androidx.compose.ui.text.buildAnnotatedString
+import androidx.compose.ui.text.font.FontFamily
+import androidx.compose.ui.text.font.FontStyle
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.BaselineShift
+import androidx.compose.ui.text.style.TextDecoration
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.Density
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.em
+import androidx.core.text.HtmlCompat
+import com.adevinta.spark.PreviewTheme
+import com.adevinta.spark.R
+import com.adevinta.spark.SparkTheme
+import com.adevinta.spark.tokens.SparkColors
+import com.adevinta.spark.tokens.SparkTypography
+
+
+// FIXME: There is no official way to do this yet so we're waiting for
+//   https://issuetracker.google.com/issues/139320238 to be fixed
+@Composable
+@ReadOnlyComposable
+public fun resources(): Resources {
+    LocalConfiguration.current
+    return LocalContext.current.resources
+}
+
+public fun Spanned.toHtmlWithoutParagraphs(): String {
+    return HtmlCompat.toHtml(this, HtmlCompat.TO_HTML_PARAGRAPH_LINES_CONSECUTIVE)
+        .substringAfter("<p dir=\"ltr\">").substringBeforeLast("</p>")
+}
+
+public fun Resources.getText(@StringRes id: Int, vararg args: Any): CharSequence {
+    val escapedArgs = args.map {
+        if (it is Spanned) it.toHtmlWithoutParagraphs() else it
+    }.toTypedArray()
+    val resource = SpannedString(getText(id))
+    val htmlResource = resource.toHtmlWithoutParagraphs()
+    val formattedHtml = String.format(htmlResource, *escapedArgs)
+    return HtmlCompat.fromHtml(formattedHtml, HtmlCompat.FROM_HTML_MODE_LEGACY)
+}
+
+@Composable
+public fun annotatedStringResource(@StringRes id: Int, vararg formatArgs: Any): AnnotatedString {
+    val resources = resources()
+    val density = LocalDensity.current
+
+    val colors = SparkTheme.colors
+    val typography = SparkTheme.typography
+    return remember(id, formatArgs) {
+        val text = resources.getText(id, *formatArgs)
+        spannableStringToAnnotatedString(text, density, colors, typography)
+    }
+}
+
+@Composable
+public fun annotatedStringResource(@StringRes id: Int): AnnotatedString {
+    val resources = resources()
+    val density = LocalDensity.current
+
+    val colors = SparkTheme.colors
+    val typography = SparkTheme.typography
+    return remember(id) {
+        val text = resources.getText(id)
+        spannableStringToAnnotatedString(text, density, colors, typography)
+    }
+}
+
+private fun spannableStringToAnnotatedString(
+    text: CharSequence,
+    density: Density,
+    colors: SparkColors,
+    typography: SparkTypography,
+): AnnotatedString {
+    return if (text is Spanned) {
+        with(density) {
```

Comment:
IMO we should not create more indentation here to the minimal benefit of having density as a receiver type.